test/celeritas/em/UrbanMsc.test.cc

```diff
@@ -273,10 +268,18 @@ TEST_F(UrbanMscTest, msc_scattering)
                                TrackStatus::alive,
                                StepLimit{}};
         sim_state_data.push_back(state);
+
+        if (i == step.size())
+        {
+            PhysicsTrackView phys = this->make_track_view(
+                pdg::electron(), MaterialId{1}, MevEnergy{energy[i]});
+            step.push_back(phys.dedx_range());
+        }
     }
     const SimStateRef& states = make_ref(states_ref);
     SimTrackView       sim_track_view(states, ThreadId{0});
 
+    EXPECT_EQ(nsamples, step.size());
```

Comment:
```suggestion
    ASSERT_EQ(nsamples, step.size());
```
to abort cleanly instead of segfaulting if the condition isn't met